packages/block-library/src/missing/edit.native.js

```diff
@@ -83,11 +97,30 @@ export class UnsupportedBlockEdit extends Component {
 				  __( "'%s' isn't yet supported on WordPress for iOS" );
 		const infoTitle = sprintf( titleFormat, title );
 
+		const actionButtonStyle = getStylesFromColorScheme(
+			styles.actionButton,
+			styles.actionButtonDark
+		);
+
 		return (
 			<BottomSheet
 				isVisible={ this.state.showHelp }
 				hideHeader
-				onClose={ this.toggleSheet.bind( this ) }
+				onClose={ this.toggleSheet }
+				onModalHide={ () => {
+					if ( this.state.sendFallbackMessage ) {
+						// On iOS, onModalHide is called when the controller is still part of the hierarchy.
+						// A small delay will ensure that the controller has already been removed.
+						this.timeout = setTimeout( () => {
+							requestUnsupportedBlockFallback(
+								attributes.originalContent,
+								clientId,
+								title
+							);
+						}, 100 );
```

Comment:
Again, it looks like others have had [issues with dismissing modals](https://github.com/react-native-community/react-native-modal/issues/192) and immediately presenting another screen, but it's hard to tell if they were fixed by using the `onModalHide` prop.

It's hard to say if 100ms will always work of if it could fail on some iOS devices. Are there any alternatives you're aware of?

Reply:
Looking into this with @etoledom there was no better solution given the situation.